src/FffGcodeWriter.cpp

```diff
@@ -2379,159 +2380,187 @@ bool FffGcodeWriter::addSupportToGCode(const SliceDataStorage& storage, LayerPla
 }
 
-
-bool FffGcodeWriter::processSupportInfill(const SliceDataStorage& storage, LayerPlan& gcode_layer) const
+bool FffGcodeWriter::processSupport(const SliceDataStorage& storage, LayerPlan& gcode_layer) const
 {
     bool added_something = false;
-    const SupportLayer& support_layer = storage.support.supportLayers[std::max(0, gcode_layer.getLayerNr())]; // account for negative layer numbers for raft filler layers
 
+    // account for negative layer numbers for raft filler layers
+    const SupportLayer& support_layer = storage.support.supportLayers[std::max(0, gcode_layer.getLayerNr())];
+
+    // Nothing to do return
     if (gcode_layer.getLayerNr() > storage.support.layer_nr_max_filled_layer || support_layer.support_infill_parts.empty())
     {
         return added_something;
     }
 
+    // create a list of outlines and use PathOrderOptimizer to optimize the travel move
+    PathOrderOptimizer<const SupportInfillPart*> island_order_optimizer(gcode_layer.getLastPlannedPositionOrStartingPosition());
+    for (const auto& part : support_layer.support_infill_parts)
+    {
+        island_order_optimizer.addPolygon(&part);
+    }
+    island_order_optimizer.optimize();
+
+    // Settings per layer Todo: maybe move out and create seperate function and setting struct
+    const bool first_layer = SupportConfig::FirstLayer(gcode_layer.getLayerNr()); // is checked multiple times for different settings
+    const bool lower_layers =  SupportConfig::LowerLayer(gcode_layer.getLayerNr()); // is checked multiple times for different settings
     const Settings& mesh_group_settings = Application::getInstance().current_slice->scene.current_mesh_group->settings;
-    const size_t extruder_nr = (gcode_layer.getLayerNr() <= 0) ? mesh_group_settings.get<ExtruderTrain&>("support_extruder_nr_layer_0").extruder_nr : mesh_group_settings.get<ExtruderTrain&>("support_infill_extruder_nr").extruder_nr;
+    const size_t extruder_nr = SupportConfig::ExtruderNr(lower_layers, mesh_group_settings);
     const ExtruderTrain& infill_extruder = Application::getInstance().current_slice->scene.extruders[extruder_nr];
 
-    coord_t default_support_line_distance = infill_extruder.settings.get<coord_t>("support_line_distance");
-    
-    // To improve adhesion for the "support initial layer" the first layer might have different properties
-    if(gcode_layer.getLayerNr() == 0)
-    {
-        default_support_line_distance = infill_extruder.settings.get<coord_t>("support_initial_layer_line_distance"); 
-    }
+    // Cluster the used support settings for easy handling and maintainability
+    SupportConfig config
+    {
+        first_layer,
+        lower_layers,
+        extruder_nr,
+        SupportConfig::LineDistance(first_layer, infill_extruder),
+        infill_extruder.settings.get<coord_t>("infill_overlap_mm"),
+        SupportConfig::InfillAngle(lower_layers, gcode_layer.getLayerNr(), storage.support),
+        1, // there is no frontend setting for this (yet)
```

Comment:
This is a "boolean trap" in integer form. I wouldn't know from reading what this `1` signifies.